include/ap/argument_parser.hpp

```diff
@@ -30,34 +42,70 @@ inline constexpr bool is_valid_type_v = is_valid_type<T, ValidTypes...>::value;
 
 
 
+/**
+ * @brief Represents the name of a command-line argument.
+ */
 struct argument_name {
+    /// @brief Default constructor (deleted).
     argument_name() = delete;
 
+    /// @brief Copy constructor.
     argument_name(const argument_name&) = default;
+
+    /// @brief Move constructor. 
     argument_name(argument_name&&) = default;
+
+    /// @brief Default copy assignment operator for argument_name.
+    /// @return Reference to the assigned argument_name.
     argument_name& operator= (const argument_name&) = default;
 
+    /// @brief Explicit parameterized constructor with only a long name. 
     explicit argument_name(const std::string_view name) : name(name) {}
+
+    /// @brief Explicit parameterized constructor with both long and short names.
     explicit argument_name(const std::string_view name, const std::string_view short_name)
         : name(name), short_name(short_name) {}
 
+    /// @brief argument_name destructor.
     ~argument_name() = default;
 
+    /**
+     * @brief Equality comparison operator for argument names.
```

Comment:
No need for "for argument names" in places like this - it's obvious that it is an operator for this class